File: java/src/com/esielkar/bridge/Abstraction.java
package com.esielkar.bridge;

public class Abstraction {
    protected Platform platform;

    public Abstraction(Platform platform) {
        this.platform = platform;
    }

    public void feature1() {
        platform.method1();
        platform.method2();
    }

    public void feature2() {
        platform.method3();
    }

}

File: dart/design_patterns/bin/bridge/abstraction.dart
import 'package:meta/meta.dart';

import 'platform.dart';

class Abstraction {
  @protected
  final Platform platform;

  Abstraction(this.platform);

  void feature1() {
    platform.method1();
    platform.method2();
  }

  void feature2() {
    platform.method3();
  }
}

class RefinedAbstraction extends Abstraction {
  RefinedAbstraction(super.platform);

  void featureN() {
    platform.method1();
    platform.method2();
    platform.method3();
  }
}
